peer/peer.go

```diff
@@ -0,0 +1,1942 @@
+// It is safe for concurrent access.
+func (p *Peer) ContinueHash() *wire.ShaHash {
+	p.relayMtx.Lock()
+	defer p.relayMtx.Unlock()
+
+	return p.continueHash
+}
+
+// SetContinueHash sets the continue hash.
+// It is safe for concurrent access.
+func (p *Peer) SetContinueHash(hash *wire.ShaHash) {
+	p.relayMtx.Lock()
+	defer p.relayMtx.Unlock()
+
+	p.continueHash = hash
+}
+
+// RelayTxDisabled returns whether or not relaying of transactions is disabled.
+// It is safe for concurrent access.
+func (p *Peer) RelayTxDisabled() bool {
+	p.relayMtx.Lock()
+	defer p.relayMtx.Unlock()
+
+	return p.disableRelayTx
+}
+
+// SetDisableRelayTx sets whether or not relaying of transactions is disabled.
+// It is safe for concurrent access.
+func (p *Peer) SetDisableRelayTx(disableRelayTx bool) {
+	p.relayMtx.Lock()
+	defer p.relayMtx.Unlock()
+
+	p.disableRelayTx = disableRelayTx
+}
+
+// LastBlock returns the last block of the peer.
+func (p *Peer) LastBlock() int32 {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.lastBlock
+}
+
+// Services returns the peer services.
+func (p *Peer) Services() wire.ServiceFlag {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.services
+}
+
+// UserAgent returns the peer user agent.
+func (p *Peer) UserAgent() string {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.userAgent
+}
+
+// LastSend returns the last send time of the peer.
+func (p *Peer) LastSend() time.Time {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.lastSend
+}
+
+// LastRecv returns the last recv time of the peer.
+func (p *Peer) LastRecv() time.Time {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.lastRecv
+}
+
+// BytesSent returns the bytes sent by the peer.
+func (p *Peer) BytesSent() uint64 {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.bytesSent
+}
+
+// BytesReceived returns the bytes received by the peer.
+func (p *Peer) BytesReceived() uint64 {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.bytesReceived
+}
+
+// TimeConnected returns the time at which the peer connected.
+func (p *Peer) TimeConnected() time.Time {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.timeConnected
+}
+
+// TimeOffset returns the time offset from the peer.
+func (p *Peer) TimeOffset() int64 {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.timeOffset
+}
+
+// StartingHeight returns the starting height of the peer.
+func (p *Peer) StartingHeight() int32 {
+	p.StatsMtx.Lock()
+	defer p.StatsMtx.Unlock()
+
+	return p.startingHeight
+}
+
+// TxProcessed sends a message that a transaction has been processed by the
+// peer.
+func (p *Peer) TxProcessed() {
+	p.txProcessed <- struct{}{}
+}
+
+// WaitForTx waits until a transaction has been processed by the peer.
+func (p *Peer) WaitForTx() {
+	<-p.txProcessed
+}
+
+// BlockProcessed sends a message that a block has been processed by the peer.
+func (p *Peer) BlockProcessed() {
```

Comment:
Same here.  Should be in the block manager struct instead of on the peer.  This is very specific to the block manager's operation and not something a generic peer should have.